azurelinuxagent/common/protocol/goal_state.py

```diff
@@ -352,35 +414,38 @@ def _parse_plugin_settings(ext_handler, plugin_settings):
         runtime_settings_node = find(settings[0], "RuntimeSettings")
         seqNo = getattrib(runtime_settings_node, "seqNo")
         runtime_settings_str = gettext(runtime_settings_node)
-        try:
-            runtime_settings = json.loads(runtime_settings_str)
-        except ValueError as e:
-            logger.error("Invalid extension settings")
-            return
+        if runtime_settings_str is not None:
+            try:
+                runtime_settings = json.loads(runtime_settings_str)
+            except ValueError as e:
+                logger.error("Invalid extension settings")
+                return
 
         depends_on_level = 0
         depends_on_node = find(settings[0], "DependsOn")
         if depends_on_node != None:
             try:
                 depends_on_level = int(getattrib(depends_on_node, "dependencyLevel"))
             except (ValueError, TypeError):
-                logger.warn("Could not parse dependencyLevel for handler {0}. Setting it to 0".format(name))
+                logger.periodic_warn(logger.EVERY_DAY,
```

Comment:
I see this message in the agent log; we should investigate whether it is related to the changes for fast track?

> WARNING ExtHandler ExtHandler Could not parse dependencyLevel for handler Microsoft.OSTCExtensions.VMAccessForLinux. Setting it to 0

Reply:
This is fixed now